### PR TITLE
Update to 0.15.0 release.

### PR DESCRIPTION
**Pull request by lbdreyer**

### CODE REVIEW

**Comment by ocefpaf** [comment]
Can you pin `matplotlib <2` too? (I guess that this version does not work properly with `matplotlib 2.0`.)

**Comment by ocefpaf** [comment]
@lbdreyer are you familiar with `conda-smithy`? If so can you rerender this in this PR? (If not don't worry I will do it later.)

**Comment by lbdreyer** [comment]
> are you familiar with conda-smithy?
@ocefpaf I believe I have used conda-smithy in the past so I hopefully should be able to rerender this feedstock

**Comment by lbdreyer** [comment]
The circle check failed: there appear to be dependency conflicts when trying to build cartopy against numpy 1.12 and Python 3.6. It looks like shapely hasn't been built for that combination. I could restrict the recipe to numpy 1.11 or Python 3.5.

Also, it looks like it is picking up matplotlib 2.0.0rc2, which I didn't realise it could do

**Comment by ocefpaf** [comment]
> Also, it looks like it is picking up matplotlib 2.0.0rc2, which I didn't realise it could do

I need to fix that in channel.

File: .travis.yml [hunk 1]
```diff
@@ -14,7 +14,6 @@ env:
     - CONDA_NPY=111  CONDA_PY=35
     - CONDA_NPY=112  CONDA_PY=35
     - CONDA_NPY=111  CONDA_PY=36
-    - CONDA_NPY=112  CONDA_PY=36
```
Comment (ocefpaf) [inline .travis.yml]: Unless you are in a hurry don't do that. I will hunt the dependencies down and get them build for Python 3.6.

**Comment by lbdreyer** [comment]
@ocefpaf I am not in a particular rush. I have reverted the last changes so now this is just waiting on shapely being built against numpy 1.12 and python 3.6, and the fix to the channel regarding the matplotlib release candidate

**Comment by ocefpaf** [comment]
> @ocefpaf I am not in a particular rush. I have reverted the last changes so now this is just waiting on shapely being built against numpy 1.12 and python 3.6, and the fix to the channel regarding the matplotlib release candidate

Perfect. I am on it.

**Comment by ocefpaf** [comment]
Argh. We don't have `matplotlib <2`  for `numpy 1.12`. This pin is really hurting our stack.

**Comment by ocefpaf** [comment]
@lbdreyer I am preparing a `matplotlib 1.*` with `numpy 1.12.*` to solve this. Hold on...

**Comment by dopplershift** [comment]
Is there really a purpose releasing this with `matplotlib < 2` set? All that's going to do is result in an install of cartopy 0.14, which doesn't have any such pin...

**Comment by ocefpaf** [comment]
> Is there really a purpose releasing this with `matplotlib < 2` set? All that's going to do is result in an install of cartopy 0.14, which doesn't have any such pin...

If someone specify `cartopy 0.15` they will get `matplotlib <2`. With this release the users have a choice.

**Pull request merged**